src/tools/rbd/action/Export.cc

```diff
@@ -552,20 +552,25 @@ static int do_export_v1(librbd::Image& image, librbd::image_info_t &info,
 }
 
 static int do_export(librbd::Image& image, const char *path, bool no_progress,
-                     int export_format)
+                     bool sync, int export_format)
 {
   librbd::image_info_t info;
   int64_t r = image.stat(info, sizeof(info));
   if (r < 0)
     return r;
 
   int fd;
+  int flag;
   int max_concurrent_ops = g_conf().get_val<uint64_t>("rbd_concurrent_management_ops");
   bool to_stdout = (strcmp(path, "-") == 0);
   if (to_stdout) {
     fd = STDOUT_FILENO;
```

Comment:
Nit: probably should fail w/ `EINVAL` if you use `--sync` when reading from stdin / writing to stdout

Reply:
Ok, I changed it. now it’s ok.